File: Leetcode/128 - Longest Consecutive Sequence/solution.py
def longestConsecutive(self, nums: List[int]) -> int:
    num_set: set[int] = set(nums)
    max_len: int = 0

    for num in num_set:
        if num - 1 not in num_set:
            curr_num: int = num
            curr_len: int = 1

            while curr_num + 1 in num_set:
                curr_num += 1
                curr_len += 1

            max_len = max(max_len, curr_len)

    return max_len

File: Leetcode/128 - Longest Consecutive Sequence/solution.cpp
int longestConsecutive(vector<int>& nums) {
    unordered_set<int> numSet(nums.begin(), nums.end());
    int maxLen = 0;
    
    for (int num: numSet) {
        // Only start counting if num is the beginning of a sequence
        if (numSet.find(num-1) == numSet.end()) {
            int currNum = num + 1, currLen = 1;
            while (numSet.find(currNum) != numSet.end()) {
                currNum++;
                currLen++;
            }
            maxLen = max(maxLen, currLen);
        }
    }

    return maxLen;
}